webrender/src/device/gfx/buffer.rs

```diff
@@ -8,10 +8,61 @@ use rendy_memory::{Block, Heaps, MemoryBlock, MemoryUsageValue, Write};
 
 use std::cell::Cell;
 use std::mem;
+use std::slice;
 
 pub const MAX_INSTANCE_COUNT: usize = 8192;
 pub const TEXTURE_CACHE_SIZE: usize = 128 << 20; // 128MB
 
+pub struct PMBuffer<B: hal::Backend> {
+    pub buffer: B::Buffer,
+    pub memory: B::Memory,
+    pub ptr: *mut u8,
+    pub coherent: bool,
+    pub height: u64,
+    pub size: u64,
+    pub state: Cell<hal::buffer::State>,
+}
+
+impl<B: hal::Backend> PMBuffer<B> {
+    pub unsafe fn acquire_host_visible_slice(&self, size: Option<u64>) -> &mut [[f32; 4]] {
```

Comment:
this should be `&mut self`, since otherwise it allows multiple mutable slices be acquired to the same thing

Reply:
Fixed in https://github.com/szeged/webrender/pull/297/commits/76b255b4b7ec31e659242e04dbd757177777abf4#diff-5dbe2f57bb9273f9a363bcb71ec6834eR26